lib/server.js

```diff
@@ -162,8 +179,14 @@ module.exports = function createServer(config, log) {
             var retryAfter = [blockEmail, blockIpEmail, blockIp].reduce(max)
             var block = retryAfter > 0
             var suspect = false
+            var blockReason = null
+
+            if (block) {
+              blockReason = blockReasons.OTHER
+            }
 
-            if (requestChecks.treatEveryoneWithSuspicion) {
+            if (requestChecks.treatEveryoneWithSuspicion
```

Comment:
Similarly you could put the reputation check part of this into the class. Something like `reputationService.isSuspectBelow(reputation)`